xmodule/docs/decisions/0003-library-content-block-schema.rst

```diff
@@ -0,0 +1,201 @@
+
+* When manual is *disabled* and shuffle is *enabled*, the block will behave as it did before. That is, it will import the entire library (filtered by capa_type) as its children, and present each user a random subset based on max_count. For backwards compatibility, these will be the default value of the settings.
+
+* When manual is *enabled*, shuffle is *disabled*, and max_count is set to -1, the block will behave in the "static" mode. That is, the user will be prompted to select specific blocks in the library, and each user will be presented those blocks in order.
+
+* When manual is *enabled*, any filter fields (currently just capa_type, but perhaps more in the future) will be ignored for the purposes of CMS deciding which blocks to import. That's because in the event that manually-picked children clash with the filters, we need to decide who would "win", and we are deciding here that the manually-picked children would win. However, we could choose to have the block-picker filter blocks based on filters.
+
+We will also remove the **mode** field, as it is no longer needed.
+
+The interaction between manual, shuffle, and max_count yields a matrix of 8 different behaviors:
+
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+|                                     | **manual = False**                                            | **manual = True**                                               |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = True,  max_count = -1** | Entire library in random order (V1's "random mode")           | Manual subset of library in random order                        |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = True,  max_count > 0**  | Random subset of library in random order (V1's "random mode") | Random subset of manual subset of library in random order       |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = False, max_count = -1** | Entire library in original order                              | Manual subset of library in original order ("static mode")      |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = False, max_count > 0**  | Random subset of library in original order                    | Random subset of manual subset of library in original order     |
```

Comment:
For "Random subset of library in original order" and "Random subset of manual subset of library in original order" I would suggest adding "(Not likely to ever be supported)", because I don't think that those use cases are ever going to be useful enough to warrant being supported. Unless you know of some compelling use case.

Reply:
We know that the latter is a desired use cases --  "Random subset of library in original order" would be (Not likely to ever be supported), but we know less about the ordering desired for "random subset of manual subset"

Reply:
@connorhaugh @bradenmacdonald How about I put "(No known use cases)" next to each of them, since we don't think either would be useful, but AFIAK we haven't checked, either.

Anyway, if we end up supporting the other six use cases, then I imagine that it would actually take more logic to forbid the user from choosing these last two than it would to just incidentally support them.